spec/ulid_spec.cr

```diff
@@ -57,4 +57,82 @@ describe ULID do
       end
     end
   end
+
+  describe ".valid?" do
+    it "should validate a valid string" do
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3").should be_true
+    end
+
+    it "should be case insensitive" do
+      ULID.valid?("01b3EAF48P97R8MP9WS6MHDTZ3").should be_true
+    end
+
+    it "should detect incorrect length" do
+      ULID.valid?("0").should be_false
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ32").should be_false
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ").should be_false
+      ULID.valid?("!@#$%^&*(").should be_false
+      ULID.valid?("abcde").should be_false
+      ULID.valid?("1234567890").should be_false
+      ULID.valid?("").should be_false
+    end
+
+    it "should detect invalid characters" do
+      ULID.valid?("01!3EAF48P97R8MP9WS8MHDTZ3").should be_false
+      ULID.valid?("01I3EAF48P97R8MP9WS8MHDTZ3").should be_false
+      ULID.valid?("01O3EAF48P97R8MP9WS8MHDTZ3").should be_false
+    end
+  end
+
+  describe ".validate!" do
+    it "should validate a valid string" do
+      ULID.validate!("01B3EAF48P97R8MP9WS6MHDTZ3").should be_nil
+    end
+
+    it "should be case insensitive" do
+      ULID.validate!("01b3EAF48P97R8MP9WS6MHDTZ3").should be_nil
+    end
+
+    it "should detect incorrect length" do
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("0") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("01B3EAF48P97R8MP9WS6MHDTZ32") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("01B3EAF48P97R8MP9WS6MHDTZ") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("!@#$%^&*(") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("abcde") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("1234567890") }
+      expect_raises(ULID::IncorrectLength) { ULID.validate!("") }
+    end
+
+    it "should detect invalid characters" do
+      expect_raises(ULID::InvalidChars) { ULID.validate!("01!3EAF48P97R8MP9WS8MHDTZ3") }
+      expect_raises(ULID::InvalidChars) { ULID.validate!("01I3EAF48P97R8MP9WS8MHDTZ3") }
+      expect_raises(ULID::InvalidChars) { ULID.validate!("01O3EAF48P97R8MP9WS8MHDTZ3") }
+    end
+  end
+
+  describe ".seed_time" do
+    it "should correctly decode seed time" do
+      seedtime = ULID.seed_time("01B3EAF48P97R8MP9WS6MHDTZ3")
+      seedtime.should be_a Time
+      seedtime.should eq Time.unix_ms(1481170718998)
+
+      seedtime1 = ULID.seed_time("01EVDRF3VB5VD3211Z4DA112V9")
+      seedtime1.should be_a Time
+      seedtime1.should eq Time.unix_ms(1610000863083)
```

Comment:
imo dont need to compare type here since Time.unix will return a Time type, it should already handle comparing type on line 121

Reply:
im being too nitty-gritty now though